pkg/ironicapi/volumes.go

```diff
@@ -1,21 +1,24 @@
 package ironicapi
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/openstack-k8s-operators/ironic-operator/pkg/ironic"
 	corev1 "k8s.io/api/core/v1"
 )
 
 // GetVolumes -
 func GetVolumes(name string) []corev1.Volume {
 	var config0640AccessMode int32 = 0640
-
+	parentName := strings.Replace(name, "-api", "", 1)
 	apiVolumes := []corev1.Volume{
```

Comment:
The other way around, we need to use GetOwningIronicName here, instead of string.Replace.